[check-webkit-style] only check a diff from git

gsnedders opened the pull request:
#### 81bfcf01ed2bbe1c1f6ae0f80b910dc657557c1a
<pre>
[check-webkit-style] only check a diff from git
<a href="https://bugs.webkit.org/show_bug.cgi?id=265030">https://bugs.webkit.org/show_bug.cgi?id=265030</a>

Reviewed by Jonathan Bedard.

Previously, we passed commit_message=True, which led us to get
git-format-tree output (an mbox, with patches formatted as a sequence of
emails with commit messages and a diffstat prior to the actual patch)
from Scm.create_patch, rather than the output of git-diff, which
DiffParser can actually parse.

Change the call to Scm.create_patch to ensure we only get a diff, and
not other metadata.

However, this exposed the fact that Git.create_patch didn&apos;t account for
the difference in command-line arguments between git-format-tree and
git-diff, thus we need to correctly pass the correct set of revisions we
wish to obtain a diff for, nor did it correctly pass the order file.

As a drive-by, add a test that we get an error from DiffParser given an
mbox.

Lastly, skip SCM tests that involve svn if it is not installed.

* Tools/Scripts/webkitpy/common/checkout/diff_parser.py:
(DiffParser._parse_into_diff_files):
* Tools/Scripts/webkitpy/common/checkout/diff_parser_unittest.py:
(DiffParserTest.setUp):
(DiffParserTest):
(DiffParserTest.tearDown):
(test_git_format_patch_multiple):
* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.create_patch):
* Tools/Scripts/webkitpy/common/checkout/scm/scm_unittest.py:
(SVNTest.setUp):
(GitTest.test_create_patch_merge_base_without_commit_message):
(GitTest):
(GitTest.test_create_patch_merge_base_with_commit_message):
(GitSVNTest.setUp):
* Tools/Scripts/webkitpy/style/main.py:
(CheckWebKitStyle.main):

Canonical link: <a href="https://commits.webkit.org/273780@main">https://commits.webkit.org/273780@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6555baeb20d557eb52071d3647131266c47a1d92

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35864 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14807 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38055 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38588 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32266 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37083 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17222 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11833 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31040 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36418 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12524 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31870 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11003 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/36059 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11043 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32041 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39838 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32589 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32365 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36949 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11177 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9079 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35039 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12927 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11687 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4740 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12008 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/webkitpy/style/main.py:
```diff
@@ -154,7 +154,7 @@ def main(self):
         else:
             changed_files = paths if options.diff_files else None
-            patch = host.scm().create_patch(options.git_commit, changed_files=changed_files, git_index=options.git_index, find_branch=True)
+            patch = host.scm().create_patch(options.git_commit, changed_files=changed_files, git_index=options.git_index, commit_message=False, find_branch=True)
```
What motivated removing the commit message from the patch?

The issue I can see with this is that we probably _do_ want to run the style checker over the commit message, right? I don't recall if our style checker knows how to parse a commit message, though...it probably doesn't so this is likely a moot point...

gsnedders replied:
It doesn't.

There are, AFAICT, two options here:

 * Make the style checker parse mboxes, and check the style for every commit message and every patch, or
 * Check the style of the total diff of all the commits.

That said, I wonder if we should make the webkitpy.common.checkout.scm.SCM.create_patch API less… odd. Because currently it returns either a unified diff or an mbox of commits, which the caller needs to parse in very different ways.

gsnedders replied:
filed https://bugs.webkit.org/show_bug.cgi?id=268403 for the API oddity

gsnedders committed:
[check-webkit-style] only check a diff from git
https://bugs.webkit.org/show_bug.cgi?id=265030

Reviewed by Jonathan Bedard.

Previously, we passed commit_message=True, which led us to get
git-format-tree output (an mbox, with patches formatted as a sequence of
emails with commit messages and a diffstat prior to the actual patch)
from Scm.create_patch, rather than the output of git-diff, which
DiffParser can actually parse.

Change the call to Scm.create_patch to ensure we only get a diff, and
not other metadata.

However, this exposed the fact that Git.create_patch didn't account for
the difference in command-line arguments between git-format-tree and
git-diff, thus we need to correctly pass the correct set of revisions we
wish to obtain a diff for, nor did it correctly pass the order file.

As a drive-by, add a test that we get an error from DiffParser given an
mbox.

Lastly, skip SCM tests that involve svn if it is not installed.

* Tools/Scripts/webkitpy/common/checkout/diff_parser.py:
(DiffParser._parse_into_diff_files):
* Tools/Scripts/webkitpy/common/checkout/diff_parser_unittest.py:
(DiffParserTest.setUp):
(DiffParserTest):
(DiffParserTest.tearDown):
(test_git_format_patch_multiple):
* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.create_patch):
* Tools/Scripts/webkitpy/common/checkout/scm/scm_unittest.py:
(SVNTest.setUp):
(GitTest.test_create_patch_merge_base_without_commit_message):
(GitTest):
(GitTest.test_create_patch_merge_base_with_commit_message):
(GitSVNTest.setUp):
* Tools/Scripts/webkitpy/style/main.py:
(CheckWebKitStyle.main):

Canonical link: https://commits.webkit.org/273780@main
